gtk/src/callbacks.c

```diff
@@ -579,6 +581,71 @@ check_name_template(signal_user_data_t *ud, const char *str)
     return FALSE;
 }
 
+static int 
+match_by_pattern(const char *string, const char *pattern)
+{
+    int status;
+    regex_t re;
+    if (regcomp(&re, pattern, REG_EXTENDED|REG_NOSUB) != 0)
+    {
+        return 0;
+    }
+    status = regexec(&re, string, (size_t) 0, NULL, 0);
+    regfree(&re);
+    if (status != 0) 
+    {
+        return 0;
+    }
+    return 1;
+}
+
+typedef struct {
+    const char *pattern;
+    const char *format;
+} datemap;
+
+static int
+parse_datestring(const char *src, struct tm *tm)
+{       
+    datemap ymdThmsZ = {"[0-9]{4}-[0-1]?[0-9]-[0-3]?[0-9]T[0-9]{2}:[0-9]{2}:[0-9]{2}Z", "%Y-%m-%dT%H:%M:%SZ"};  
+
+    datemap maps[1] = { ymdThmsZ };
+
+    for (int i = 0; i < sizeof(maps); i++)
+    {
+        if (match_by_pattern(src, maps[i].pattern))
+        {
+            strptime(src, maps[i].format, tm);
```

Comment:
Fix compiler warning implicit declaration strptime
This is a posix extension that is not part of the C standard.  It needs a special define _XOPEN_SOURCE to enable it in header file parsing.

This can be fixed by adding:
AC_USE_SYSTEM_EXTENSIONS
to gtk/configure.ac above 'AC_PROG_CC' line.